### DISCUSSION TITLE
Error: is not a function

### DISCUSSION THREAD

**Discussion by oschwarzenberg** [alpinejs/alpine#2553] [1. Help]
If I try to run this snippet from codesandbox https://codesandbox.io/s/pintura-and-alpine-sitting-in-a-tree-viicc locally, i only get a blank screen. if i change `async init()` to `init()` i get this error: `(index):27 Uncaught TypeError: appendDefaultEditor is not a function`.

What is wrong in my code or local configuration. the sandbox runs and is from the author of pintura.

```html
<!DOCTYPE html>

<link rel="stylesheet" href="./pintura.css" />

<div
     x-data="pintura"
     x-init="src = './my-image.jpeg'"
     x-on:load="e => console.log('image loaded', e.detail)"
     x-on:process="e => console.log('image processed', e.detail)"></div>

<script src="//unpkg.com/alpinejs" defer></script>
<script>
  document.addEventListener("alpine:init", () => {
    // define Pintura component
    Alpine.data("pintura", () => ({
      // image source
      src: undefined,

      // init editor
      async init() {
        // load editor module
        const { appendDefaultEditor, dispatchEditorEvents } = await import(
          "./pintura.js"
        );

        // init with current src
        const editor = appendDefaultEditor(this.$el, {
          // initial props
          src: this.src,

          // route editor events to node
          handleEvent: (type, detail) => {
            this.$dispatch(type, detail);
          }
        });

        // observe src prop
        this.$watch("src", () => {
          // load new src
          editor.src = this.src;
        });
      }
    }));
  });
</script>
```

**Comment by SimoTod** [comment]
You can't remove async because your code needs await to load the function. Hard to say without seeing the code but I think you just don't have pintura.js or it's not where your code says it should be (same level as your html page). I don't know that library but be aware that the file says you need a licence to use it on your domain.

**Reply by oschwarzenberg** [reply]
I have a legal licence of pintura. It's the same like in the codesandbox except the copyright note. All files are in the same directory. So I think it's the same structure as in the codesanbox. I get also the same error with my legal licence and with the demo licence.

**Reply by SimoTod** [reply]
That's really odd. Try to put a try catch into the async function and print out the error in console, it should give you more info. If the error is still the same, it means that it can't find the js file for some reason.
